src/scales.js

```diff
@@ -105,3 +116,22 @@ function inferScaleType(key, channels, {type, domain, range}) {
 function asOrdinalType(key) {
   return registry.get(key) === position ? "point" : "ordinal";
 }
+
+// prepare scales for exposure through the plot's scales() function
+export function exposeScales(scaleDescriptors) {
+  return Object.fromEntries(
+    Object.entries(scaleDescriptors)
+    .map(([key, descriptor]) => [key, exposeScale(descriptor)])
+  );
+}
+
+function exposeScale({scale, label}) {
+  return {
+    domain: scale.domain(),
+    range: scale.range(),
+    ...scale.interpolate && {interpolate: scale.interpolate()},
+    ...label !== undefined && {label},
+    ...scale.type && {type: scale.type},
+    ...scale.clamp && scale.clamp() && {clamp: true}
```

Comment:
```suggestion
    ...scale.clamp && {clamp: scale.clamp()}
```

Reply:
When do we want to have the {clamp: false} option exposed, since it's the default?

Reply:
I think the exposed scale should materialize the defaults, not leave them unspecified, unless you want to specifically distinguish between the case where the user specified `clamp: false` versus when they didn’t specify the clamp option and it defaulted to false.

Reply:
OK. My design choice was because I was aiming for the terser option, i.e. return the minimal options object that defines the scale. I don't think we can address the difference between explicit and default false at this point, since the option has been "folded in" the scale.

Reply:
Philosophically speaking, it is good to be minimal on input, but unambiguous on output, and hence we want to materialize the defaults.